x/deployment/keeper/keeper_test.go

```diff
@@ -211,6 +211,35 @@ func Test_OnDeploymentClosed(t *testing.T) {
 	})
 }
 
+func Test_CloseGroup(t *testing.T) {
+	ctx, keeper := setupKeeper(t)
+	groups := createActiveDeployment(t, ctx, keeper)
+
+	t.Run("assert group 0 state closed", func(t *testing.T) {
+		assert.NoError(t, keeper.OnCloseGroup(ctx, groups[0]))
+		group, ok := keeper.GetGroup(ctx, groups[0].ID())
+		assert.True(t, ok)
+		assert.Equal(t, types.GroupClosed, group.State)
+	})
+	t.Run("group 1 matched-state not-orderable", func(t *testing.T) {
+		group := groups[1]
+		t.Logf("%v", types.GroupStateToReadable(group.State))
+		assert.Equal(t, group.ValidateOrderable(), types.ErrGroupNotOpen)
```

Comment:
let's keep the printf debugging out of commits.

I like how this is checking the _behavior_ -- "is it orderable?" -- instead of the precise state.  That said, in this case we want to ensure that the state hasn't changed (checking accidental closing of all groups).  `OnCloseGroup(ctx,groups[0])` doesn't affect whether or not `groups[1]` is orderable (it isn't orderable before and it isn't orderable after).

Reply:
The reason neither group are 'open', is that `createActiveDeployment()` initializes two groups in state 'matched', so they're not "Orderable" from the start. 

I've create an additional test to assert the case you outlined of closing one group and not affecting sibling groups.

Reply:
> The reason neither group are 'open', is that createActiveDeployment() initializes two groups in state 'matched', so they're not "Orderable" from the start.

exactly, hence

> it isn't orderable before and it isn't orderable after